dooit: `2.2.0` -> `3.0.2`

kraanzu opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kraanzu commented:
@khaneliman , I've enabled tests as well but lemme know if anything can be done in a better way :)

khaneliman commented:
Looks like formatting got broken

kraanzu commented:
Yes pretty weird. I'm using `alejandra` for formatting and it doesn't complain.
I tried again with `nixfmt` but that also failed :?

khaneliman commented:
We use nixfmt-rfc-style

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Murli Tawari ***@***.***>
Sent: Sunday, November 10, 2024 9:52:02 AM
To: NixOS/nixpkgs ***@***.***>
Cc: Austin Horstman ***@***.***>; Mention ***@***.***>
Subject: Re: [NixOS/nixpkgs] dooit: `2.2.0` -> `3.0.1` (PR #354955)


Yes pretty weird. I'm using alejandra for formatting and it doesn't complain.
I tried again with nixfmt but that also failed :?

—
Reply to this email directly, view it on GitHub<https://github.com/NixOS/nixpkgs/pull/354955#issuecomment-2466785940>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AANSH3QEYSF2M52575QPX53Z756KFAVCNFSM6AAAAABRQBO5EGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINRWG44DKOJUGA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

kraanzu committed:
dooit: `2.2.0` -> `3.0.2`

Update pkgs/by-name/do/dooit/package.nix

Co-authored-by: Austin Horstman <khaneliman12@gmail.com>

Update pkgs/by-name/do/dooit/package.nix

Co-authored-by: Austin Horstman <khaneliman12@gmail.com>

Update pkgs/by-name/do/dooit/package.nix

Co-authored-by: Austin Horstman <khaneliman12@gmail.com>

Update pkgs/by-name/do/dooit/package.nix

Co-authored-by: Austin Horstman <khaneliman12@gmail.com>

changes according to sugesstions

format with nixfmt

Update pkgs/by-name/do/dooit/package.nix

Co-authored-by: Austin Horstman <khaneliman12@gmail.com>

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
 python3.pkgs.buildPythonApplication rec {
-  pname = "dooit";
-  version = "2.2.0";
+  pname = repo;
```
```suggestion
  pname = "dooit";
```
https://github.com/nix-community/nixpkgs-lint/issues/21

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
-    rev = "v${version}";
-    hash = "sha256-GtXRzj+o+FClleh73kqelk0JrSyafZhf847lX1BiS9k=";
+    owner = username;
+    repo = pname;
```
```suggestion
    repo = "dooit";
```
^

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
-    hash = "sha256-GtXRzj+o+FClleh73kqelk0JrSyafZhf847lX1BiS9k=";
+    owner = username;
+    repo = pname;
+    rev = ver;
```
```suggestion
  rev = "refs/tags/v${version}";
```

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
   python3 = python311;
+  username = "dooit-org";
+  repo = "dooit";
+  ver = "v3.0.2";
```
```suggestion
  ver = "3.0.2";
```

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -64,8 +73,8 @@ python3.pkgs.buildPythonApplication rec {
 
   meta = with lib; {
     description = "TUI todo manager";
-    homepage = "https://github.com/kraanzu/dooit";
-    changelog = "https://github.com/kraanzu/dooit/blob/v${version}/CHANGELOG.md";
+    homepage = "https://github.com/${username}/${pname}";
+    changelog = "https://github.com/${username}/${pname}/blob/v${ver}/CHANGELOG.md";
```
Would like to undo these. There's a lot of discussions around why it might be more DRY but is troublesome in nixpkgs.

kraanzu replied:
Hmm..didn't know, I'll undo it. Why is this troublesome though?

khaneliman replied:
Changing meta attributes based on package overrides, i believe.

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
   python311,
   testers,
   nix-update-script,
+  extraPackages ? [ ],
```
What is the purpose of introducing this?

kraanzu replied:
So the config file for dooit is in python and dooit now supports plugins which wont be possible unless the libraries that are being used are in the same space? See [Installation](https://dooit-org.github.io/dooit/getting_started/installation.html#flakes)

Also I mailed you regarding this. I think it got lost :)

khaneliman replied:
Oh, shoot... I think I remember seeing a notification for that a long time ago... sorry. I'll take a look closer at that new functionality.

kraanzu replied:
No problemo! I eventually got that working. It was so frustrating but it worked out in the end :)

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
+    export HOME=$(mktemp -d)
+  '';
+
+  nativeCheckInputs = with python3.pkgs; [
+    pytest
+    faker
+  ];
+  doCheck = true;
+  checkPhase = ''
+    python -m pytest
+  '';
```
Do we need all this or could we get away with `pytestCheckHook`?

kraanzu replied:
Didn't know about this. I'll check it out

khaneliman commented:
We should also separate refactoring / cleanup commits from version bumps.

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
 
   src = fetchFromGitHub {
-    owner = "kraanzu";
+    owner = username;
```
```suggestion
    owner = "dooit-org";
```

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
 }:
 let
   python3 = python311;
+  username = "dooit-org";
+  repo = "dooit";
+  ver = "3.0.2";
```
```suggestion
```

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,62 @@
 python3.pkgs.buildPythonApplication rec {
   pname = "dooit";
-  version = "2.2.0";
+  version = ver;
```
```suggestion
  version = "3.0.2";
```

kraanzu commented:
Made all the suggested changes

khaneliman commented on pkgs/by-name/do/dooit/package.nix:
```diff
@@ -5,53 +5,54 @@
+      pyperclip
+      textual
+      pyyaml
+      dateutil
```
```suggestion
      python-dateutil
```

khaneliman reviewed (approved):
LGTM, thanks

kraanzu commented:
hi @khaneliman ! Thanks for merging, I wanted to ask if I should create the `dooit-extras` package in the same directory?

Since it's `/pkgs/by-name/do/...` or should it be different? Thanks :)

khaneliman commented:
> hi @khaneliman ! Thanks for merging, I wanted to ask if I should create the `dooit-extras` package in the same directory?
> 
> Since it's `/pkgs/by-name/do/...` or should it be different? Thanks :)

Yeah, it'd go in there as `/do/dooit-extras` with out by-name setup.
